src/activity_handlers.cpp

```diff
@@ -845,41 +842,43 @@ void activity_handlers::pulp_do_turn( player_activity *act, player *p )
 
 void activity_handlers::refill_vehicle_do_turn( player_activity *act, player *p )
 {
-    vehicle *veh = NULL;
-    veh = g->m.veh_at(act->placement.x, act->placement.y);
-    if( !veh ) {  // Vehicle must've moved or something!
+    vehicle *veh = g->m.veh_at( act->placement );
+    if( veh == nullptr ) {  // Vehicle must've moved or something!
         act->moves_left = 0;
         return;
     }
     bool fuel_pumped = false;
-    for(int i = -1; i <= 1; i++ ) {
-        for(int j = -1; j <= 1; j++ ) {
-            if( g->m.ter(p->posx() + i, p->posy() + j) == t_gas_pump ||
```

Comment:
This loops looks around the player character position (`p->pos`), the new code looks around the vehicles position (the position of the selected vehicle is stored as activity placement).

The following setup shows the bug:
```
  &
 @
v
```
`v` is a vehicle, the new code looks around it and does not fine the pump which is **2** squares away.